exchanges/binance/binance_wrapper.go

```diff
@@ -836,47 +1262,205 @@ func (b *Binance) GetOrderHistory(req *order.GetOrdersRequest) ([]order.Detail,
 	if err := req.Validate(); err != nil {
 		return nil, err
 	}
-
 	if len(req.Pairs) == 0 {
 		return nil, errors.New("at least one currency is required to fetch order history")
 	}
-
 	var orders []order.Detail
-	for x := range req.Pairs {
-		resp, err := b.AllOrders(req.Pairs[x],
-			"",
-			"1000")
-		if err != nil {
-			return nil, err
-		}
-
-		for i := range resp {
-			orderSide := order.Side(strings.ToUpper(resp[i].Side))
-			orderType := order.Type(strings.ToUpper(resp[i].Type))
-			// New orders are covered in GetOpenOrders
-			if resp[i].Status == "NEW" {
-				continue
-			}
-
-			pair, err := currency.NewPairFromString(resp[i].Symbol)
+	switch req.AssetType {
+	case asset.Spot, asset.Margin:
+		for x := range req.Pairs {
+			resp, err := b.AllOrders(req.Pairs[x],
+				"",
+				"1000")
 			if err != nil {
 				return nil, err
 			}
 
-			orders = append(orders, order.Detail{
-				Amount:   resp[i].OrigQty,
-				Date:     resp[i].Time,
-				Exchange: b.Name,
-				ID:       strconv.FormatInt(resp[i].OrderID, 10),
-				Side:     orderSide,
-				Type:     orderType,
-				Price:    resp[i].Price,
-				Pair:     pair,
-				Status:   order.Status(resp[i].Status),
-			})
+			for i := range resp {
+				orderSide := order.Side(strings.ToUpper(resp[i].Side))
+				orderType := order.Type(strings.ToUpper(resp[i].Type))
+				// New orders are covered in GetOpenOrders
+				if resp[i].Status == "NEW" {
+					continue
+				}
+
+				pair, err := currency.NewPairFromString(resp[i].Symbol)
+				if err != nil {
+					return nil, err
+				}
+
+				orders = append(orders, order.Detail{
+					Amount:   resp[i].OrigQty,
+					Date:     resp[i].Time,
+					Exchange: b.Name,
+					ID:       strconv.FormatInt(resp[i].OrderID, 10),
+					Side:     orderSide,
+					Type:     orderType,
+					Price:    resp[i].Price,
+					Pair:     pair,
+					Status:   order.Status(resp[i].Status),
+				})
+			}
+		}
+	case asset.CoinMarginedFutures:
+		for i := range req.Pairs {
+			switch {
+			case !req.StartTicks.IsZero() && !req.EndTicks.IsZero() && req.OrderID == "":
+				if req.EndTicks.After(req.StartTicks) {
+					if time.Since(req.StartTicks) > time.Hour*24*30 {
```

Comment:
This has been brought up earlier, but no explanation was given on the PR itself, so sadly I have to bring it up. This looks to be saying if > 30 days, return error, only 7 days worth of data can be retrieved. What's going on since this was resolved?

Reply:
Pretty sure I just conversed with shazbert directly. Sorry for not writing up a response here. Will do that moving forward to save the hassle. The deal here is that binance allow 7days lookup for usdtmargined and 30days for coinmargined

Reply:
Thanks for that! Please update the error message to say 30 days then